draft-kinnear-webtransport-http2.md

```diff
@@ -63,459 +78,248 @@ Discussion of this draft takes place on the WebTransport mailing list
 
-                                    SETTINGS
-                                    SETTINGS_ENABLE_CONNECT_[..] = 1
-                                    SETTINGS_ENABLE_WEBTRANSPORT = 1
+   Pad Length:  An 8-bit field containing the length of the frame
+      padding in units of octets.  This field is conditional (as
+      signified by a "?" in the diagram) and is only present if the
+      PADDED flag is set.
 
-HEADERS + END_HEADERS
-+ STREAM_ID = 3
-:method = CONNECT
-:protocol = webtransport
-:scheme = https
-:path = /
-:authority = server.example.com
+   Session ID:  An unsigned 31-bit integer that identifies the
+      stream Connect Stream for this Web Transport stream.  The
+      Session ID MUST be MUST be an open stream negotiated via the
+      extended CONNECT protocol with a `:protocol` value of
+      "webtransport".
 
-                                    HEADERS + END_HEADERS
-                                    + STREAM_ID = 3
-                                    :status = 200
+   Data:  Application data.  The amount of data is the remainder of the
+      frame payload after subtracting the length of the other fields
+      that are present.
 
-                                    WTHEADERS + END_HEADERS
-                                    + STREAM_ID = 2
-                                    + CONNECT_STREAM = 3
-                                    :method = GET
-                                    :scheme = https
-                                    :path = /
-                                    :authority = client.example.com
+The WT_DATAGRAM frame defines the following flags:
 
-WTHEADERS + END_HEADERS
-+ STREAM_ID = 2
-+ CONNECT_STREAM = 3
-:status = 200
+   UNIDIRECTIONAL (0x1):  When set, the stream begins in the
+      "half-closed (remote)" state.
 
-                                    DATA + STREAM_ID = 2
-                                    WebTransport Data
+The recipient MAY respond with a stream error of type WT_STREAM_ERROR if the
+specified WebTransport Connect Stream does not exist, is not a stream
+established via extended CONNECT to use the "webtransport" protocol, or if it is
+in the "closed" or "half-closed (remote)" stream state.
 
-DATA + STREAM_ID = 2 + END_STREAM
-WebTransport Data
+The data in WT_DATAGRAM frames is not subject to flow control.  The receiver MAY
+discard this data if it does not have sufficient space to buffer it.
 
-                                    DATA + STREAM_ID = 2 + END_STREAM
-                                    WebTransport Data
-~~~
+An intermediary could forward the data in a WT_DATAGRAM frame over another
+protocol, such as WebTransport over HTTP/3.  In QUIC, a datagram frame can span
+at most one packet.  Because of that, the applications have to know the maximum
+size of the datagram they can send.  However, when proxying the datagrams, the
+hop-by-hop MTUs can vary.
 
+TODO: Describe how the path MTU can be computed, specifically propagation across
+HTTP proxies.
```

Comment:
Can we move this to an issue and keep it out of the ID?
```suggestion
```

Reply:
Also came from h3, but will remove from here.